### PR TITLE
[#170] Created an endpoint for submitting hashtags the user is interested in

### LINKED ISSUE #170: Create an endpoint for submitting hashtags the user is interested in
**Is your feature request related to a problem? Please describe.**
Related to [issue 151](https://github.com/SE701Team2/Updog/issues/151) and [issue 157](https://github.com/SE701Team2/Updog/issues/157).

**Describe the solution you'd like**
The endpoint requires an array of hashtag names as well as the user id. 

Create a table called UserInterests in the form of
(PK) ID: INTEGER
(FK) USER_ID INTEGER
(FK) TAG_ID: INTEGER
Constraint: Unique(USER_ID, TAG_ID)

Add these tags for the user id into the table.

**Describe alternatives you've considered**
The endpoint could accept tag ids instead of names which would mean we would not have to query the tag table in the database to find the TAG_ID.

### PR DESCRIPTION
**Pull request by DanilLinkov**
# Description:
Closes #170

Changes:
 - Created new table userInterests
 - Created interests controller with GET, PUT and DELETE method handlers
 - Routed the methods in routes/index.js to GET, PUT, DELETE /api/interests
 - Created tests for the new interests controller
 - Updated swagger documentation

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: backend/server/routes/index.js [hunk 1]
```diff
@@ -19,6 +19,12 @@ router.route('/users/authenticate').post(user.authenticateUser)
 
 router.route('/users/:username/activity').get(user.getUserActivity)
 
+router.route('/interests').get(interests.getInterests)
```
Comment (f-fromont) [inline backend/server/routes/index.js]: Nit: If you could group these like how the other endpoints do

**Review by f-fromont — APPROVED** [review]
LGTM after the small nit is resolved

**Comment by f-fromont** [comment]
Also if you could add the issue in brackets to the pr as well! [#196 ]

File: backend/server/routes/index.js [hunk 2]
```diff
@@ -19,6 +19,12 @@ router.route('/users/authenticate').post(user.authenticateUser)
...
+
+router.route('/interests').post(interests.addInterests)
+
+router.route('/interests').delete(interests.deleteInterests)
+
```
Comment (scorpionknifes) [inline backend/server/routes/index.js]:
```suggestion
router
  .route('/interests')
  .get(interests.getInterests)
  .post(interests.addInterests)
  .delete(interests.deleteInterests)

```
example (not sure if this is prettier)

**Review by scorpionknifes — CHANGES_REQUESTED** [review]

**Review by colaMeowSY — APPROVED** [review]
LGTM！

**Review by f-fromont — APPROVED** [review]
LGTM

**Review by scorpionknifes — APPROVED** [review]

**Pull request merged**